### PR TITLE
Add documentation to partial.py.

### PR DESCRIPTION
**Pull request by dhatch**

### CODE REVIEW

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py [hunk 1]
```diff
@@ -78,8 +153,28 @@ def translate_and(expression: Expression, session: Session, model, get_model):
 
     return expr
 
-
 def translate_isa(expression: Expression, session: Session, model, get_model):
+    """Translate an Isa operation. (matches keyword)
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py:157]:
```suggestion
    """Translate an Isa operation. (``matches`` keyword)
```

Unsure if it's supposed to be 1 or 2 backticks, but looks like you're using 2 everywhere else.

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py [hunk 2]
```diff
@@ -94,12 +189,28 @@ def translate_isa(expression: Expression, session: Session, model, get_model):
...
-
 def translate_compare(expression: Expression, session: Session, model, get_model):
+    """Translate a binary comparison operation.
+
+    Operators are listed in ``COMPARISONS``.
+
+    Either the left or right argument may contain a path. Paths for both
+    arguments (i.e. post.name = post.body) are not supported currently.
+
+    Also handle _this = ? where ? is an instance of the same type as _this.
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py:200]:
```suggestion
    Also handle unification of _this with an instance of the same type as _this. E.g., _this = ? where ? is an instance of the same type as _this.
```

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py [hunk 3]
```diff
@@ -94,12 +189,28 @@ def translate_isa(expression: Expression, session: Session, model, get_model):
...
+
+    _this.path.(path1)+.tail OP val => Model.path.(path1)+.has(Target.tail OP val)
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py:202]: I don't really understand this notation

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py [hunk 4]
```diff
@@ -110,13 +221,16 @@ def translate_compare(expression: Expression, session: Session, model, get_model
...
+    # Dot operation is on right
     elif right_path and right_path[0] == "_this":
         return translate_compare(
             Expression(flip_op(expression.operator), [right, left]),
             session,
             model,
             get_model,
         )
+    # this = other no dot operation, throws if it's not of the form _this = other other same type as
+    # this
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py:233]: Might need to clean this up.

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py [hunk 5]
```diff
@@ -138,6 +252,18 @@ def translate_compare(expression: Expression, session: Session, model, get_model
 
 
 def translate_in(expression, session, model, get_model):
+    """Translate the in operator.
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py:255]: What is "relationship" referring to here?

File: languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py [hunk 6]
```diff
@@ -178,6 +304,20 @@ def translate_in(expression, session, model, get_model):
 
 
 def translate_dot(path: Tuple[str, ...], session: Session, model, func: EmitFunction):
+    """Translate an operation over a path.
```
Comment (leina05) [inline languages/python/sqlalchemy-oso/sqlalchemy_oso/partial.py:307]: Could you describe each of the params?

**Review by leina05 — CHANGES_REQUESTED** [review]
These are really helpful docs, thanks Dave. Left a few comments, I know the hairier methods are harder to explain in docstrings.

**Comment by dhatch** [comment]
@leina05 I don't really have time to update this but I'd like to merge as is if you're okay with it. I think some documentation is helpful even if it's not ideal.

**Review by leina05 — APPROVED** [review]

**Pull request merged**